pkg/scheduler/core/generic_scheduler.go

```diff
@@ -271,8 +271,15 @@ func (g *genericScheduler) processPreemptionWithExtenders(
 			if extender.SupportsPreemption() {
 				var err error
 				// Replace nodeToVictims with result after preemption from extender.
-				if nodeToVictims, err = extender.ProcessPreemption(pod, nodeToVictims, g.cachedNodeInfoMap); err != nil {
-					return nil, err
+				if nodeToVictims, err = extender.ProcessPreemption(
```

Comment:
When ProcessPreemption returns an error, it returns nil for `nodeToVictims`. This would cause preemption to not happen. You should not overwrite `nodeToVictims` when the extender is ignorable. If it returns error, you should bypass the extender and pass the existing `nodeToVictims` to the next extender.

Reply:
Thanks for pointing out, will fix it.

Reply:
@resouer It would be great to add a test for this, i.e., having two extenders that support preemption, one of them is ignorable and returns error, the second one removes one more node  from the 'nodeToVictims' map and then test that preemption logic works as expected.

Reply:
Of course, test added, thank you for reviewing.